compiler/passes/src/code_generation/program.rs

```diff
@@ -231,16 +233,13 @@ impl<'a> CodeGenerator<'a> {
         // Helper to construct the string associated with the type.
         let create_type = |type_: &Type| {
             match type_ {
-                Type::Mapping(_) | Type::Tuple(_) => unreachable!("Mappings cannot contain mappings or tuples."),
+                Type::Mapping(_) | Type::Tuple(_) => panic!("Mappings cannot contain mappings or tuples."),
                 Type::Identifier(identifier) => {
                     // Lookup the type in the composite mapping.
                     // Note that this unwrap is safe since all struct and records have been added to the composite mapping.
                     let (is_record, _) = self.composite_mapping.get(&identifier.name).unwrap();
-                    match is_record {
-                        // If the type is a struct, then add the public modifier.
-                        false => self.visit_type_with_visibility(type_, Mode::Public),
-                        true => unreachable!("Type checking guarantees that mappings cannot contain records."),
-                    }
+                    assert!(!is_record, "Type checking guarantees that mappings cannot contain records.");
```

Comment:
Based on our standards discussion, shouldn't we be using a `panic!` here?

Reply:
I thought so, at least in the `unreachable` vs `panic` discussion.
IMO `assert` implies a less strong of a failure than a `panic`. More of a style preference.
Ultimately the error will get surfaced to the user as an exception.

I don't have strong opinions except for wanting to nail down the standard.
If you're down, we can make a call right here and I'll throw it into the standards tracking issue.

Reply:
I think its reasonable to use either but it would be helpful to have a (minimal) set of rules for our codebase.
Last we discussed, we decided not to use `unreachable!` 
Do we want the rule to be allow either `panic!` or `assert!`?